mautic/mautic
# [UX] Dynamic content slot generator

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

Did you already feel a desire to give up on using dynamic content because the feature doesn't help you to make it usable?

this PR fixes the need of reading the docs and crafting manually, by hand, your dynamic content slot for inserting into a website.

Also:
- fixes alerts positioned by absolute instead of fixed (you weren't able to see it if scrolling down the page before)
- migrated some classes to alternatives to ensure UI is defining them to the right elements
- fixed a bug on the buttons line height for sizes bigger than lg (they were misaligned)
- text-muted class became text-secondary, and I hope other devs keep it consistent (fallback still there for compatibility)




| Before                                 | After
| -------------------------------------- | ---
|        ![image](https://github.com/user-attachments/assets/881d6d35-66b6-4742-a4fe-6920b59494c7)                                | ![image](https://github.com/user-attachments/assets/eca95747-ec16-4c50-8d47-8af1a092c8d5)

![image](https://github.com/user-attachments/assets/d4e61c8c-4f0e-44ea-b301-429ab0e8de5e)

![image](https://github.com/user-attachments/assets/106341c8-0c36-4557-bc3d-46e28e3bba97)




---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Dynamic Content
3. Create new
4. Set Is campaign based? to No
5. Define a name for the slot
6. Save & close
7. On its details, check the generator
8. Write anything as default content and click to copy
9. See that you receive a success message
10. Try changing wrapper or using the HTML mode, click copy and paste somewhere to see it
